diplomdb/src/main/java/at/ac/tgm/diplomarbeit/diplomdb/controller/DokumentController.java

```diff
@@ -0,0 +1,414 @@
+
+import java.io.IOException;
+import java.nio.file.*;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.stream.Collectors;
+
+/**
+ * Controller zur Verwaltung von Dokumenten.
+ *
+ * Dieser Controller stellt Endpunkte zur Verfügung, um Dokumente hochzuladen, herunterzuladen,
+ * abzurufen, zu aktualisieren, zu bewerten und zu löschen. Die Funktionalitäten sind für die Rollen
+ * STUDENT, TEACHER und ADMIN zugänglich, wobei bestimmte Aktionen (wie die Bewertung) zusätzlichen
+ * Berechtigungen unterliegen.
+ *
+ * Verfügbare Endpunkte:
+ * - POST /api/documents/upload: Upload eines neuen Dokuments.
+ * - GET /api/documents/download/{id}: Download eines Dokuments anhand seiner ID.
+ * - GET /api/documents: Abruf einer Liste von Dokumenten mit optionaler Filterung und Sortierung.
+ * - GET /api/documents/{id}: Abruf eines spezifischen Dokuments anhand seiner ID.
+ * - PUT /api/documents/{id}: Aktualisierung der Dokumentendaten.
+ * - PUT /api/documents/{id}/rating: Aktualisierung der Bewertung eines Dokuments.
+ * - DELETE /api/documents/{id}: Löschen eines Dokuments und der zugehörigen Datei.
+ */
+@RestController
+@RequestMapping("api/documents")
+public class DokumentController {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DokumentController.class);
+    private static final Logger AUDIT_LOGGER = LoggerFactory.getLogger("at.ac.tgm.diplomarbeit.diplomdb.audit");
+
+    @Autowired
+    private DokumentRepository dokumentRepository;
+
+    @Autowired
+    private DiplomarbeitRepository diplomarbeitRepository;
+
+    @Autowired
+    private UserService userService;
+
+    private static final String UPLOAD_DIR = "uploads/";
+
+    /**
+     * Erlaubt den Upload eines Dokuments.
+     *
+     * HTTP-Methode: POST
+     * URL: /api/documents/upload
+     *
+     * @param file Die hochzuladende Datei.
+     * @param titel Der Titel des Dokuments.
+     * @param beschreibung Eine Beschreibung des Dokuments.
+     * @param typ Der Typ des Dokuments (z.B. Lastenheft, Design-Dokument).
+     * @param datum Das Datum des Dokuments im Format YYYY-MM-DD.
+     * @param diplomarbeitId Die ID der zugehörigen Diplomarbeit.
+     * @param erstellerSamAccountName Der sAMAccountName des Erstellers.
+     * @return ResponseEntity mit dem erstellten Dokument-Datentransferobjekt.
+     */
+    @Secured({Roles.STUDENT, Roles.TEACHER, Roles.ADMIN})
+    @PostMapping("/upload")
+    public ResponseEntity<DokumentDTO> uploadDocument(
+            @RequestParam("file") MultipartFile file,
+            @RequestParam("titel") String titel,
+            @RequestParam("beschreibung") String beschreibung,
+            @RequestParam("typ") String typ,
+            @RequestParam("datum") String datum,
+            @RequestParam("diplomarbeitId") Long diplomarbeitId,
+            @RequestParam("erstellerSamAccountName") String erstellerSamAccountName) {
```

Comment:
Name kann weggelassen werden, wenn gleich mit dem Variablennamen

```suggestion
            @RequestParam String erstellerSamAccountName) {
```